dio/lib/src/dio_event_processor.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:async';
+
+import 'package:dio/dio.dart';
+import 'package:sentry/sentry.dart';
+// ignore: implementation_imports
+import 'package:sentry/src/sentry_exception_factory.dart';
+
+/// This is an [EventProcessor], which improves crash reports of [DioError]s.
+/// It adds information about [DioError.response] if present and also about
+/// the inner exceptions.
+class DioEventProcessor implements EventProcessor {
+  /// This is an [EventProcessor], which improves crash reports of [DioError]s.
+  DioEventProcessor(this._options, this._maxRequestBodySize);
+
+  final SentryOptions _options;
+  final MaxRequestBodySize _maxRequestBodySize;
+
+  SentryExceptionFactory get _sentryExceptionFactory =>
+      // ignore: invalid_use_of_internal_member
+      _options.exceptionFactory;
```

Comment:
Maybe if we use `_options.exceptionFactory` directly, we avoid this linter ignore.

Reply:
Nvm, the problem would still exist, Since its `@internal`, the linter only likes it if used within its own package.